packages/core/src/types.ts

```diff
@@ -16,12 +20,31 @@ export interface IPrismConfig {
   validateResponse: boolean;
 }
 
-export type ValidatorFn<Resource, T> = (opts: { resource: Resource; element: T }) => IPrismDiagnostic[];
+export type ValidatorFn<Resource, T> = (opts: { resource?: Resource; element: T }) => IPrismDiagnostic[];
```

Comment:
If we're validating, both the resource to validate against and the element must be present. What'd be the use case for `resource` to be undefined?